CMakeLists.txt

```diff
@@ -12,6 +11,7 @@ set(CMAKE_OBJCOPY arm-none-eabi-objcopy)
 set(CMAKE_OBJDUMP arm-none-eabi-objdump)
 set(SIZE arm-none-eabi-size)
 set(CMAKE_TRY_COMPILE_TARGET_TYPE STATIC_LIBRARY)
+set(CMAKE_EXPORT_COMPILE_COMMANDS TRUE)
```

Comment:
same here, the compile_commands.json is put in the build dir (which my IDE picks up), so it doesn't really clutter anything

Reply:
This is fine, it might give a warning if you use a CMake generator that does not support generating a compile_commands.json file (only Makefiles and Ninja support it), but I can't see when anyone would use any others.